### PR TITLE
DROOLS-5263: Wrong test result status if rules have not been fired

### PR DESCRIPTION
**Pull request by yesamer**
@danielezonca @dupliaka Can you please test and review it?

Basically, an empty data fact on GIVEN section, now acts like the fact is not present at all.

Thank you @danielezonca  for the hint.

https://issues.redhat.com/browse/DROOLS-5263
https://issues.redhat.com/browse/RHDM-1330

### CODE REVIEW

File: drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/runner/RuleScenarioRunnerHelperTest.java [hunk 1]
```diff
@@ -240,6 +240,14 @@ public void extractExpectedValuesTest() {
         List<ScenarioExpect> scenario2Outputs = runnerHelper.extractExpectedValues(scenario2.getUnmodifiableFactMappingValues());
         assertEquals(3, scenario2Outputs.size());
         assertEquals(1, scenario2Outputs.stream().filter(ScenarioExpect::isNewFact).count());
+
+        /* A Given "TEST" fact with null rawValue should works as the previous case, i.e. to not consider the GIVEN fact with empty data */
+        scenario2.addOrUpdateMappingValue(FactIdentifier.create("TEST", String.class.getCanonicalName()),
```
Comment (dupliaka) [inline drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/runner/RuleScenarioRunnerHelperTest.java:245]: How about move that code section in a separate function and reuse it?
assertGivenFactWithEmptyDataIsNotConsidering()

**Review by danielezonca — APPROVED** [review]

**Comment by dupliaka** [comment]
- [x] If user did not initialized an object then section cell in EXPECT get marked as red and error popup contains message that instance on this condition was not found.
- [x] Initialized object passes tests

**Review by dupliaka — APPROVED** [review]
Looks good by me

**Comment by yesamer** [comment]
@danielezonca can you please merge it?

**Pull request merged**